data/netcdf_data.py

```diff
@@ -569,7 +578,27 @@ def latlon_variables(self):
         )
 
     @property
-    def depth_dimensions(self):
+    def dimensions(self) -> List[str]:
```

Comment:
How we're doing the dimensions here feels a little code-smelly...to call this you need an instance of this class which opens the database or hits the url at start, but this implementation is re-opening all the stuff anyways.

We can address this at a later date so this is good for now.